src/pool/srv_pool.c

```diff
@@ -2259,29 +2367,35 @@ ds_pool_list_cont_handler(crt_rpc_t *rpc)
 	if (rc != 0)
 		D_GOTO(out, rc);
 
-	rc = rdb_tx_begin(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term, &tx);
-	if (rc != 0)
-		D_GOTO(out_svc, rc);
+	/* Verify pool handle only if RPC initiated by a client
+	 * (not for mgmt svc to pool svc RPCs that do not have a handle).
+	 */
+	crt_req_src_rank_get(rpc, &srcrank);
+	if (srcrank == ((d_rank_t) -1)) {
```

Comment:
Seems like something that might benefit from a const or some helper method. Guess we'd have to add that in CaRT though.

Reply:
There is a constant in CaRT. I had hoped to use it, but it's not exposed to callers unfortunately. I don't want to update the cart used in daos in this PR however (big hammer, typically causes other bigger problems). I could file a cart ticket if we think it's necessary in the future to resolve.

Reply:
If we're doing these checks with more commands (which seems likely - dmg and daos tool will share some CaRT calls like this), I think it'd be nice to file a CaRT ticket so we can use that constant in the future.

Reply:
OK a CaRT PR is in progress to move the CRT_NO_RANK definition into the API https://github.com/daos-stack/cart/pull/309